test/integration/xds_integration_test.cc

```diff
@@ -597,6 +597,59 @@ TEST_P(LdsIntegrationTest, NewListenerWithBadPostListenSocketOption) {
   test_server_->waitForCounterGe("listener_manager.listener_create_failure", 1);
 }
 
+// Verify the grpc cached logger is available after the initial logger filter is destroyed.
+TEST_P(LdsIntegrationTest, GrpcLoggerSurvivesAfterReloadConfig) {
+  autonomous_upstream_ = true;
+  const std::string grpc_logger_string = R"EOF(
+    name: grpc_accesslog
+    typed_config:
+      "@type": type.googleapis.com/envoy.extensions.access_loggers.grpc.v3.HttpGrpcAccessLogConfig
+      common_config:
+        log_name: bar
+        transport_api_version: V3
+        grpc_service:
+          envoy_grpc:
+            cluster_name: cluster_0
+  )EOF";
+
+  config_helper_.addConfigModifier(
+      [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) { TestUtility::loadFromYaml(grpc_logger_string, *hcm.add_access_log()); });
+  initialize();
+  // Given we're using LDS in this test, initialize() will not complete until
+  // the initial LDS file has loaded.
+  EXPECT_EQ(1, test_server_->counter("listener_manager.lds.update_success")->value());
+
+  // HTTP 1.0 is disabled by default.
+  std::string response;
+  sendRawHttpAndWaitForResponse(lookupPort("http"), "GET / HTTP/1.0\r\n\r\n", &response, true);
+  EXPECT_TRUE(response.find("HTTP/1.1 426 Upgrade Required\r\n") == 0);
+
+  test_server_->waitForCounterGe("access_logs.grpc_access_log.logs_written", 1);
+
+  // Create a new config with HTTP/1.0 proxying.
+  ConfigHelper new_config_helper(
+      version_, *api_, MessageUtil::getJsonStringFromMessageOrDie(config_helper_.bootstrap()));
+  new_config_helper.addConfigModifier(
+      [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) {
+        hcm.mutable_http_protocol_options()->set_accept_http_10(true);
+        hcm.mutable_http_protocol_options()->set_default_host_for_http_10("default.com");
+        TestUtility::loadFromYaml(grpc_logger_string, *hcm.add_access_log());
+      });
+
+  // Create an LDS response with the new config, and reload config.
+  new_config_helper.setLds("1");
+  test_server_->waitForCounterGe("listener_manager.listener_in_place_updated", 1);
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 2);
+
+  // HTTP 1.0 should now be enabled.
+  std::string response2;
+  sendRawHttpAndWaitForResponse(lookupPort("http"), "GET / HTTP/1.0\r\n\r\n", &response2, false);
+  EXPECT_THAT(response2, HasSubstr("HTTP/1.0 200 OK\r\n"));
+  test_server_->waitForCounterGe("access_logs.grpc_access_log.logs_written", 2);
```

Comment:
Shouldn't we use `waitForCounterEq()` here and at L628? I was thinking about the case when the counter is 2 at L628 already and the logger doesn't get used for some reason.

Reply:
I agree. I think `Eq` is better here; there shouldn't ever be additional logs written out accidentally during this test.